### ISSUE TITLE
NC22 Doctrine\DBAL\Query\QueryException: More than 1000 expressions in a list are not allowed on Oracle

### ISSUE THREAD

**Issue by toriva** [nextcloud/groupfolders#1702]
Good day! I hope I posted the message correctly.
Please help. After NC auto-update via snap to version 22, an error appeared and synchronization works intermittently. Please tell me what to do? Thank you in advance.

### Server configuration

**Operating system:** Ubuntu 20.04.2 LTS x86_64

**Web server:** Apache (fpm-fcgi)

**Database:** mysql 8.0.26

**PHP version:** 8.0.11

Modules loaded: Core, date, libxml, openssl, pcre, zlib, bcmath, bz2, ctype, curl, dom, hash, fileinfo, filter, ftp, gd, gmp, SPL, iconv, intl, json, ldap, mbstring, pcntl, PDO, session, posix, Reflection, standard, SimpleXML, mysqlnd, exif, tokenizer, xml, xmlreader, xmlwriter, zip, pdo_mysql, cgi-fcgi, redis, Zend OPcache

**Nextcloud version:** 22.2.0 - 22.2.0.2

**Group folders version:** 10.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** auto-updated via snap

**Are you using encryption:** no

### Logs

```
Doctrine\DBAL\Query\QueryException: More than 1000 expressions in a list are not allowed on Oracle.
/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/RuleManager.php - line 113:
OC\DB\QueryBuilder\QueryBuilder->execute()
/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLManager.php - line 77:
OCA\GroupFolders\ACL\RuleManager->getRulesForFilesByPath()
/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLManager.php - line 118:
OCA\GroupFolders\ACL\ACLManager->getRules()
/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLCacheWrapper.php - line 98:
OCA\GroupFolders\ACL\ACLManager->preloadPaths()
/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLCacheWrapper.php - line 68:
OCA\GroupFolders\ACL\ACLCacheWrapper->preloadEntries()
/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php - line 103:
OCA\GroupFolders\ACL\ACLCacheWrapper->getFolderContentsById()
/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php - line 103:
OC\Files\Cache\Wrapper\CacheWrapper->getFolderContentsById()
/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php - line 103:
OC\Files\Cache\Wrapper\CacheWrapper->getFolderContentsById()
/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php - line 103:
OC\Files\Cache\Wrapper\CacheWrapper->getFolderContentsById()
/snap/nextcloud/28586/htdocs/lib/private/Files/View.php - line 1461:
OC\Files\Cache\Wrapper\CacheWrapper->getFolderContentsById()
/snap/nextcloud/28586/htdocs/apps/dav/lib/Connector/Sabre/Directory.php - line 264:
OC\Files\View->getDirectoryContent()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Tree.php - line 200:
OCA\DAV\Connector\Sabre\Directory->getChildren()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 900:
Sabre\DAV\Tree->getChildren()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 982:
Sabre\DAV\Server->generatePathNodes()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 1661:
Sabre\DAV\Server->getPropertiesIteratorForPath()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 1646:
Sabre\DAV\Server->writeMultiStatus()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/CorePlugin.php - line 346:
Sabre\DAV\Server->generateMultiStatus()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/event/lib/WildcardEmitterTrait.php - line 89:
Sabre\DAV\CorePlugin->httpPropFind()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 472:
Sabre\DAV\Server->emit()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 253:
Sabre\DAV\Server->invokeMethod()
/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php - line 321:
Sabre\DAV\Server->start()
/snap/nextcloud/28586/htdocs/apps/dav/lib/Server.php - line 333:
Sabre\DAV\Server->exec()
/snap/nextcloud/28586/htdocs/apps/dav/appinfo/v2/remote.php - line 35:
OCA\DAV\Server->exec()
/snap/nextcloud/28586/htdocs/remote.php - line 166:
require_once("/snap/nextc ... p")
```

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
`{"reqId":"zMlRrbEZ7bw5hu9W39qf","level":3,"time":"2021-10-22T08:30:26+00:00","remoteAddr":"192.168.43.21","user":"moderator","app":"core","method":"PROPFIND","url":"/remote.php/dav/files/moderator/%D0%9B%D0%BE%D0%BA%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%B5%20%D0%B7%D0%B0%D0%BF%D0%B8%D1%81%D0%B8","message":"More than 1000 expressions in a list are not allowed on Oracle.","userAgent":"Mozilla/5.0 (Windows) mirall/3.3.5stable-Win64 (build 20210930) (Nextcloud, windows-10.0.19042 ClientArchitecture: x86_64 OsArchitecture: x86_64)","version":"22.2.0.2","exception":{"Exception":"Doctrine\\DBAL\\Query\\QueryException","Message":"More than 1000 expressions in a list are not allowed on Oracle.","Code":0,"Trace":[{"file":"/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/RuleManager.php","line":113,"function":"execute","class":"OC\\DB\\QueryBuilder\\QueryBuilder","type":"->"},{"file":"/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLManager.php","line":77,"function":"getRulesForFilesByPath","class":"OCA\\GroupFolders\\ACL\\RuleManager","type":"->"},{"file":"/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLManager.php","line":118,"function":"getRules","class":"OCA\\GroupFolders\\ACL\\ACLManager","type":"->"},{"file":"/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLCacheWrapper.php","line":98,"function":"preloadPaths","class":"OCA\\GroupFolders\\ACL\\ACLManager","type":"->"},{"file":"/var/snap/nextcloud/28586/nextcloud/extra-apps/groupfolders/lib/ACL/ACLCacheWrapper.php","line":68,"function":"preloadEntries","class":"OCA\\GroupFolders\\ACL\\ACLCacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php","line":103,"function":"getFolderContentsById","class":"OCA\\GroupFolders\\ACL\\ACLCacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php","line":103,"function":"getFolderContentsById","class":"OC\\Files\\Cache\\Wrapper\\CacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php","line":103,"function":"getFolderContentsById","class":"OC\\Files\\Cache\\Wrapper\\CacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/lib/private/Files/Cache/Wrapper/CacheWrapper.php","line":103,"function":"getFolderContentsById","class":"OC\\Files\\Cache\\Wrapper\\CacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/lib/private/Files/View.php","line":1461,"function":"getFolderContentsById","class":"OC\\Files\\Cache\\Wrapper\\CacheWrapper","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/apps/dav/lib/Connector/Sabre/Directory.php","line":264,"function":"getDirectoryContent","class":"OC\\Files\\View","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Tree.php","line":200,"function":"getChildren","class":"OCA\\DAV\\Connector\\Sabre\\Directory","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":900,"function":"getChildren","class":"Sabre\\DAV\\Tree","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":982,"function":"generatePathNodes","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":1661,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":1646,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/apps/dav/lib/Server.php","line":333,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/snap/nextcloud/28586/htdocs/remote.php","line":166,"args":["/snap/nextcloud/28586/htdocs/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/snap/nextcloud/28586/htdocs/lib/private/DB/QueryBuilder/QueryBuilder.php","Line":268,"CustomMessage":"More than 1000 expressions in a list are not allowed on Oracle."}}`
</details>
<details>
<summary>List of activated apps</summary>
Enabled:

- accessibility: 1.8.0
- activity: 2.15.0

 - admin_audit: 1.12.0
 - bruteforcesettings: 2.2.0
 - circles: 22.1.1
 - cloud_federation_api: 1.5.0
 - comments: 1.12.0
 - contactsinteraction: 1.3.0
 - dashboard: 7.2.0
 - dav: 1.19.0
 - federatedfilesharing: 1.12.0
 - federation: 1.12.0
 - files: 1.17.0
 - files_accesscontrol: 1.12.1
 - files_automatedtagging: 1.12.0
 - files_pdfviewer: 2.3.0
 - files_rightclick: 1.1.0
 - files_sharing: 1.14.0
 - files_trashbin: 1.12.0
 - files_versions: 1.15.0
 - files_videoplayer: 1.11.0
 - firstrunwizard: 2.11.0
 - groupfolders: 10.0.0
 - logreader: 2.7.0
 - lookup_server_connector: 1.10.0
 - nextcloud_announcements: 1.11.0
 - notifications: 2.10.1
 - oauth2: 1.10.0
 - password_policy: 1.12.0
 - photos: 1.4.0
 - privacy: 1.6.0
 - provisioning_api: 1.12.0
 - recommendations: 1.1.0
 - serverinfo: 1.12.0
 - settings: 1.4.0
 - sharebymail: 1.12.0
 - support: 1.5.0
 - survey_client: 1.10.0
 - systemtags: 1.12.0
 - text: 3.3.0
 - twofactor_backupcodes: 1.11.0
 - user_ldap: 1.12.0
 - user_status: 1.2.0
 - viewer: 1.6.0
 - weather_status: 1.2.0
 - workflowengine: 2.4.0

Disabled:
 - encryption
 - files_external
 - theming
</details>

<details>
<summary>Configuration (config/config.php)</summary>

{
    "apps_paths": [
        {
            "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
            "url": "\/extra-apps",
            "writable": true
        }
    ],
    "supportedDatabases": [
        "mysql"
    ],
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0
    },
    "log_type": "file",
    "logfile": "\/var\/snap\/nextcloud\/current\/logs\/nextcloud.log",
    "logfilemode": 416,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "192.168.12.34"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "22.2.0.2",
    "overwrite.cli.url": "http:\/\/192.168.12.34",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "loglevel": 3,
    "maintenance": false,
    "ldapIgnoreNamingRules": false,
    "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory"
}

**Comment by twojtyniak** [comment]
This seems to be a showstopper for me. Synchronizations never complete since the "upgrade" making the system unusable for purpose. How do I solve by sorting by folders? Any other workarounds? Is this something that is going to be addressed?

**Comment by acsfer** [comment]
Fixed by https://github.com/nextcloud/groupfolders/pull/1804

**Comment by twojtyniak** [comment]
Thank you for taking care of this issue. Perhaps this is not the right place for this comment/question, but I really got hurt by this bug and am trying to figure out how to keep things safe. I lost data, credibility with others, and faith in NC because of this. I used NC with GF for a couple of years, have recommended it to others, and recently implemented it for a client. It was fantastic until this "stable" "update" release.

My inclination is to not keep "current" even though the releases are supposed to be "stable", but the release cycle for NC is so fast, things seem to get left behind quickly. At this point I have not re-implemented NC for the client after doing my best to recover the data. Because of the way this bug hit, there was no way to get the multiple clients to finish syncing, so no way to know what is the most-current, correct state of a large number of files and not possible to manually verify things are recovered now. We have had to just move on. GF really provides powerful tools, but this bug has spooked me.

As an enterprise software architect and developer for 25 years I feel that I have some perspective on bugs and the fallibility of software. What I am hoping is that the testing protocols have changed because of this bug. I guess I am wondering what would give me confidence that this devastating level of bug will not get through into another "stable" release. This is an honest question. Do I need to run parallel servers and do my own testing for every release?

I have offered to be of assistance within my interest and skill set, but have not had much of a response besides "Fork the code and make the fix." I could use a bit more hand-holding to understand how things work within this project, but would love to contribute and help to make the product better and stable enough that it becomes more practical for more users to implement and manage.
